### PR TITLE
[CALCITE] Added support for CREATE TABLE attribute FREESPACE

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: babel/src/main/codegen/includes/parserImpls.ftl [hunk 1]
```diff
@@ -236,6 +236,25 @@ SqlCreateAttribute CreateTableAttributeJournalTable() :
     { return new SqlCreateAttributeJournalTable(id, getPos()); }
 }
 
+// FREESPACE attribute can take in decimals but should truncate to an integer
```
Comment (AndrewPochapsky) [inline babel/src/main/codegen/includes/parserImpls.ftl:239]: nit: punctuation.

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 2]
```diff
@@ -388,6 +388,36 @@ private void checkParseInfixCast(String sqlType) {
...
+    final String expected = "(?s).*Encountered \"-\" at .*";
+    sql(sql).fails(expected);
+  }
+
+  @Test public void testTableAttributeFreeSpaceOutOfRangeFails() {
+    final String sql = "create table foo, freespace = ^82.5^ (bar integer)";
+    final String expected = "(?s).*Numeric literal.*out of range.*";
+    sql(sql).fails(expected);
+  }
+
```
Comment (dstekol) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:420]: Could also add failure case for percent option.

File: babel/src/main/codegen/includes/parserImpls.ftl [hunk 3]
```diff
@@ -236,6 +236,25 @@ SqlCreateAttribute CreateTableAttributeJournalTable() :
...
+SqlCreateAttribute CreateTableAttributeFreeSpace() :
+{
+    SqlLiteral tempNumeric;
+    int freeSpaceValue;
+    boolean percent = false;
+}
+{
+    <FREESPACE> <EQ> tempNumeric = UnsignedNumericLiteral() {
+        freeSpaceValue = tempNumeric.getValueAs(Integer.class);
+        if (freeSpaceValue < 0 || freeSpaceValue > 75) {
```
Comment (ArthurInTheShell) [inline babel/src/main/codegen/includes/parserImpls.ftl:249]: It would be nice to have edge case tests, for example the value equals 0, -1, 75.
